backend/debug/reg_with_debug_info.mli

```diff
@@ -76,22 +76,22 @@ val assigned_to_stack : t -> bool
 
 val clear_debug_info : t -> t
 
-module Set_distinguishing_names_and_locations : Set.S with type elt = t
-
-module Map_distinguishing_names_and_locations : Map.S with type key = t
-
 module Set : sig
   include Set.S with type elt = t
 
+  val print : Format.formatter -> t -> unit
+
   val of_array : reg_with_debug_info array -> t
 
   val mem_reg : t -> Reg.t -> bool
+  val mem_reg_by_loc : t -> Reg.t -> bool
 
   val find_reg_exn : t -> Reg.t -> reg_with_debug_info
 
   val find_reg_with_same_location_exn : t -> Reg.t -> reg_with_debug_info
 
   val filter_reg : t -> Reg.t -> t
+  val filter_reg_by_loc : t -> Reg.t -> t
```

Comment:
`filter_reg` is unused now (enable warnings?). consider renaming `filter_reg_by_loc` to something like `remove_reg_by_loc` to be consistent with the positive meaning of filtering on sets.